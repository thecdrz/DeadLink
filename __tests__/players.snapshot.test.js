const { playersListEmbed } = require('../lib/embeds');

describe('players embed snapshot', () => {
  test('playersListEmbed with sample players', () => {
    const players = [
      { name: 'Alice', level: 12, health: 92, kills: 5, deaths: 0, ping: 45, sessionDistance: 120, location: 'Near the central hub', sessionDuration: '15m', streak: 60 },
      { name: 'Bob', level: 5, health: 76, kills: 1, deaths: 2, ping: 120, sessionDistance: 10, location: 'Outskirts', sessionDuration: '5m', streak: 5 }
    ];
  const embed = playersListEmbed({ players });
  // Normalize dynamic footer for deterministic snapshots
  if (embed.footer) embed.footer.text = 'Generated on <fixed>'; 
  expect(embed).toMatchSnapshot();
  });
});
