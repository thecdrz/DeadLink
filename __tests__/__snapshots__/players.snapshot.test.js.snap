// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`players embed snapshot playersListEmbed with sample players 1`] = `
{
  "color": 3066993,
  "fields": [
    {
      "inline": false,
      "name": "Alice L12 — ❤️ 92",
      "value": "⚔️ 5 · ☠️ 0 · 45ms · 15m · 📏 120m · 🔥 60m
↳ Near the central hub",
    },
    {
      "inline": false,
      "name": "Bob L5 — ❤️ 76",
      "value": "⚔️ 1 · ☠️ 2 · 120ms · 5m · 📏 10m · 🔥 5m
↳ Outskirts",
    },
  ],
  "footer": {
    "text": "Generated on <fixed>",
  },
  "title": "👥 Current Players Online",
}
`;
